.github/workflows/npm-cd.yml

```diff
@@ -69,12 +73,53 @@ jobs:
                           arch: arm64,
                           TARGET: aarch64-apple-darwin,
                       }
+                    - {
+                          OS: ubuntu,
+                          NAMED_OS: linux,
+                          ARCH: arm64,
+                          TARGET: aarch64-unknown-linux-musl,
+                          RUNNER: [self-hosted, Linux, ARM64],
+                          IMAGE: 'node:alpine',
+                          CONTAINER_OPTIONS: '--user root --privileged --rm',
+                      }
+                    - {
+                          OS: ubuntu,
+                          NAMED_OS: linux,
+                          ARCH: x64,
+                          TARGET: x86_64-unknown-linux-musl,
+                          RUNNER: ubuntu-latest,
+                          IMAGE: 'node:alpine',
+                          CONTAINER_OPTIONS: '--user root --privileged',
+                      }
         steps:
             - name: Setup self-hosted runner access
-              if: ${{ contains(matrix.build.RUNNER, 'self-hosted') }}
+              if: ${{ contains(matrix.build.RUNNER, 'self-hosted') && matrix.build.TARGET != 'aarch64-unknown-linux-musl' }}
               run: sudo chown -R $USER:$USER /home/ubuntu/actions-runner/_work/glide-for-redis
 
+            - name: Install dependencies
```

Comment:
Can you move `Install dependencies`, `Clean repository for musl on ARM64`, `Set up access for musl on ARM` amd `Setup node`  to a different action (e.g `setup-linux-musl`), and use this action here and in the build-and-test-linux-musl-latest job? it will both remove code duplication and reduce the number of steps we use here only for musl to aa single one

Reply:
In order to make it clean ill move just what is needed for musl on ARM since it is all for ARM but the installation and then i dont need if statement in the file.

Reply:
Apparently we has to be as i said since composite in an action file demands and we dont have bash before installing dependencies' so i have to leave the installing outside of the action file